### PR TITLE
Add builders for arm64 on macOS unopt

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/122534

Adds a descritiption for building the host_debug_unopt_arm64 and ios_sim_unopt_arm64 to the mac arm64 bots.

Does not run host tests yet, but will be easy to do that after https://github.com/flutter/flutter/issues/103386 is resolved.

@cbracken fyi

### CODE REVIEW

File: ci/builders/mac_unopt.json [hunk 1]
```diff
@@ -111,6 +111,109 @@
...
+            ]
+        },
+        {
+            "archives": [
+                {
+                    "base_path": "out/host_debug_unopt_arm64/zip_archives/",
+                    "type": "gcs",
+                    "include_paths": [
+                    ],
+                    "name": "host_debug_unopt_arm64"
```
Comment (godofredoc) [inline ci/builders/mac_unopt.json:123]: Is the only purpose of this build to validate it builds correctly?

File: ci/builders/mac_unopt.json [hunk 2]
```diff
@@ -111,6 +111,109 @@
...
+                }
+            ],
+            "drone_dimensions": [
+                "device_type=none",
+                "os=Mac-12",
+                "cpu=arm64"
+            ],
+            "gclient_custom_vars": {
```
Comment (godofredoc) [inline ci/builders/mac_unopt.json:131]: gclient_variables?

**Review by jmagman — APPROVED** [review]
cc @keyonghan this will add more arm builders that potentially could be xcode-cache corrupted if we try to update just the arm bots to test out your named cache work. However we can turn these off temporarily if that's a problem, if needed.

LGTM once @godofredoc is happy.

**Comment by godofredoc** [comment]
Adding my +2 with nits because I'll be ooo the rest of the day.

**Review by godofredoc — APPROVED** [review]
LGTM with nits

File: ci/builders/mac_unopt.json (hunk 1 above)
Comment (dnfield) [reply ci/builders/mac_unopt.json:123]: For now, until we can run tests.

File: ci/builders/mac_unopt.json (hunk 2 above)
Comment (dnfield) [reply ci/builders/mac_unopt.json:131]: This was copy-pasted from the other ones.

File: ci/builders/mac_unopt.json (hunk 1 above)
Comment (dnfield) [reply ci/builders/mac_unopt.json:123]: I should have said:

We're not ready to run host unit tests on CI for reasons in https://github.com/flutter/flutter/issues/103386. It seems close.

This is running tests on iOS simulator though.

**Pull request merged**